packages/babel-parser/test/helpers/run-fixture-tests.js

```diff
@@ -0,0 +1,189 @@
+          `${name}/${title}`,
+          test,
+          onlyCompareErrors,
+        );
+      }
+    }
+  }
+}
+
+function runAutogeneratedParseTests(
+  parse,
+  prefix,
+  task,
+  onlyCompareErrors = false,
+) {
+  const { expect, options } = task;
+  const testFn = task.disabled ? it.skip : it;
+
+  const expected = deserialize(expect.loc, options, expect.code);
+  const title = `${prefix}/${task.title}`;
+  const toStartPosition = ({ startLine = 1, startColumn = 0 }) =>
+    `(${startLine}, ${startColumn})`;
+
+  toFuzzedOptions(options)
+    .map(([adjust, options], index) => ({
+      ...task,
+      title: `${title} start = ${toStartPosition(options)}`,
+      adjust,
+      options,
+      expected,
+      filename: task.actual.loc,
+      source: task.actual.code,
+      original: index === 0,
+    }))
+    .forEach(test =>
+      testFn(test.title, () => runParseTest(parse, test, onlyCompareErrors)),
+    );
+}
+
+const toJustErrors = result => ({
+  threw: result.threw,
+  ast: result.ast && { errors: result.ast.errors },
+});
+
+function runParseTest(parse, test, onlyCompareErrors) {
+  const { adjust, expected, source, filename, options } = test;
+
+  if (expected.threw && expected.ast) {
+    throw Error(
+      "File expected.json exists although options specify throws. Remove expected.json.",
+    );
+  }
+
+  const actual = parseWithRecovery(parse, source, filename, options);
+  const difference = new Difference(
+    adjust,
+    onlyCompareErrors ? toJustErrors(expected) : expected,
+    onlyCompareErrors ? toJustErrors(actual) : actual,
+  );
+
+  // No differences means we passed and there's nothing left to do.
+  if (difference === Difference.None) return;
+
+  const error = FixtureError.fromDifference(difference, actual);
+
+  // If we're not overwriting the current values with whatever we get this time
+  // around, then we have a legitimate error that we need to report.
+  if (CI || !OVERWRITE) throw error;
+
+  // We only write the output of the original test, not all it's auto-generated
+  // variations.
+  if (!test.original) return;
+
+  const testLocation = test.taskDir;
+
+  // FIXME: We're just maintaining the legacy behavior of storing *just* the
+  // error `message` here, which differs from the error's `toString()` that we
+  // store for each error in the `errors` array. In both cases, we should
+  // serialize the full error to be able to property test locations,
+  // reasonCodes, etc.
+  const throws = !!actual.threw && actual.threw.message;
+  const optionsLocation = join(testLocation, "options.json");
+
+  // We want to throw away the contents of `throws` here.
+  // eslint-disable-next-line no-unused-vars
+  const { throws: _, ...oldOptions } = readJSON(optionsLocation);
+  const newOptions = { ...oldOptions, ...(throws && { throws }) };
+
+  // Store (or overwrite) the options file if there's anything to record,
+  // otherwise remove it.
+  if (Object.keys(newOptions).length > 0) {
```

Comment:
I don't think we need to update `options.json` if `options.json` does not contains `throws` at all, i.e.

https://github.com/babel/babel/blob/8ccd1219f830784ead6178180f87fd23e0bf9bd9/packages/babel-parser/test/fixtures/estree/options.json#L1-L3

I notice this issue because prettier and `JSON.stringify` have different output here.